### DISCUSSION TITLE
Performance Issues

### DISCUSSION THREAD

**Discussion by tambarskjelve** [kaveh808/kons-9#192] [Q&A] [answered]
Hi, newcomer to the Common Lisp ecosystem here, and follows this project with great interest. After some trial and error kons-9 compiles and runs without error, but the application may have performance issues. The fans starts almost immediately spinning, and the keyboard gets warm when the application launches. SBCL shows continous 100-120% CPU-usage in Activity Monitor when the application is running. SBCL and glfw are installed from MacPorts, cl-confidence and kons-9 are manually downloaded, everything else is installed from Quicklisp. 

*Describe output*
`Uname: Darwin
System: macOS 13.0.1
Architecture: x86_64
Date: Mon Nov 21 11:25:34 CET 2022
SBCL: /opt/local/bin/sbcl (SBCL 2.2.10)
Brew: :NOT-FOUND
MacPorts: /opt/local/bin/port
QuickLisp: T`

*Testsuite output*
`Name: KONS-9/TESTSUITE:RUN-ALL-TESTS
Total: 43
Success: 43/43 (100%)
Failure: 0/43 (0%)
Condition: 0/43 (0%)
Outcome: Success`

**Comment by kaveh808** [comment]
Greetings and welcome to the project.

On my system (MacOS 11.6.4) sbcl uses up 100%+ of the CPU when running `test-demos.lisp` but settles down to about 25% when idle. The CPU temperature is around 52C.

**Comment by ghost** [comment]
On Ubuntu 20.04, I had similar results to kaveh808 with ```test-demos.lisp```--it is an underpowered laptop and the fan didn't even spin up.

**Comment by tambarskjelve** [comment]
Is there any debugging tool I can use to investigate kons-9s cpu-usage?

**Comment by ghost** [comment, marked as answer]
> debugging tool 

SBCL has code profiling support:

https://www.sbcl.org/manual/#Profiling

**Reply by tambarskjelve** [reply]
Thanks I'll try to figure that out

**Reply by foretspaisibles** [reply]
See also #195 (no additional information here but worth noting).
